Penteract/Assets/Scripts/HUDManager.cpp

```diff
@@ -0,0 +1,1210 @@
+			GameObject* pictoShade = children[(static_cast<int>(cooldown)) % 3]->GetChild(HIERARCHY_INDEX_ABILITY_PICTO_SHADE);
+
+			if (pictoShade->HasChildren()) {
+				GameObject* pictoFillObj = pictoShade->GetChild(static_cast<unsigned int>(0));
+				if (pictoFillObj) {
+					ComponentImage* pictoFill = pictoFillObj->GetComponent<ComponentImage>();
+					if (pictoFill) {
+						pictoFill->SetColor(colorToUse);
+					}
+				}
+			}
+
+		}
+	} else {
+		//Switch ability picto state
+		if (children.size() < HIERARCHY_INDEX_SWITCH_ABILITY_PICTO_SHADE) return;
+
+		GameObject* pictoShade = children[HIERARCHY_INDEX_SWITCH_ABILITY_PICTO_SHADE];
+		if (pictoShade) {
+			if (pictoShade->HasChildren()) {
+				GameObject* pictoFill = pictoShade->GetChild(static_cast<unsigned int>(0));
+				if (pictoFill) {
+					ComponentImage* pictoFillImage = pictoFill->GetComponent<ComponentImage>();
+					if (pictoFillImage) {
+						pictoFillImage->SetColor(colorToUse);
+					}
+				}
+			}
+		}
+	}
+
+
+}
+
+void HUDManager::GetAllHealthColors() {
+	if (!fangHealthParent || !onimaruHealthParent) return;
+	if (fangHealthChildren.size() != 5 || onimaruHealthChildren.size() != 5) return;
+
+	// Get background color in background
+	ComponentImage* image = onimaruHealthChildren[HIERARCHY_INDEX_HEALTH_BACKGROUND]->GetComponent<ComponentImage>();
+	if (image) healthBarBackgroundColorInBackground = image->GetColor();
+
+	// Get fill color in background
+	image = onimaruHealthChildren[HIERARCHY_INDEX_HEALTH_FILL]->GetComponent<ComponentImage>();
+	if (image) healthFillBarColorInBackground = image->GetColor();
+
+	// Get overlay color in background
+	image = onimaruHealthChildren[HIERARCHY_INDEX_HEALTH_OVERLAY]->GetComponent<ComponentImage>();
+	if (image) healthOverlayColorInBackground = image->GetColor();
+
+	// Get health text color in background
+	ComponentText* healthText = onimaruHealthChildren[HIERARCHY_INDEX_HEALTH_TEXT]->GetComponent<ComponentText>();
+	if (healthText) healthTextColorInBackground = healthText->GetFontColor();
+
+	// Get main background color
+	image = fangHealthChildren[HIERARCHY_INDEX_HEALTH_BACKGROUND]->GetComponent<ComponentImage>();
+	if (image) healthBarBackgroundColor = image->GetColor();
+
+	// Get main fill color
+	image = fangHealthChildren[HIERARCHY_INDEX_HEALTH_FILL]->GetComponent<ComponentImage>();
+	if (image) healthFillBarColor = image->GetColor();
+
+	// Get main overlay color
+	image = fangHealthChildren[HIERARCHY_INDEX_HEALTH_OVERLAY]->GetComponent<ComponentImage>();
+	if (image) healthOverlayColor = image->GetColor();
+
+	// Get main health text color
+	healthText = fangHealthChildren[HIERARCHY_INDEX_HEALTH_TEXT]->GetComponent<ComponentText>();
+	if (healthText) healthTextColor = healthText->GetFontColor();
+
+
+	// Get lost feedback colors
+	image = fangHealthChildren[HIERARCHY_INDEX_HEALTH_LOST_FEEDBACK]->GetComponent<ComponentImage>();
+	if (image) {
+		Debug::Log("Hiiiiii");
```

Comment:
Remove log

Reply:
Yep

Reply:
Done

Reply:
I haven't pointed every Debug::Log, because it would be quite spammy, but yeah, .cpp must be reviewed and every log deleted, but that can be done later on